### ISSUE TITLE
"could not decorrelate subquery" when Yii2 gets column info

### ISSUE THREAD

**Issue by Sarke** [cockroachdb/cockroach#85595]
**Describe the problem**

I get a `SQLSTATE[XXUUU]: <<Unknown error>>: 7 ERROR: could not decorrelate subquery` error.

The code in question is from https://github.com/yiisoft/yii2/blob/master/framework/db/pgsql/Schema.php#L473-L533

Specifically, the issue is with this part, as removing it removes the error:

```sql
    CASE WHEN COALESCE(td.typtype, tb.typtype, t.typtype) = 'e'::char
        THEN array_to_string((SELECT array_agg(enumlabel) FROM pg_enum WHERE enumtypid = COALESCE(td.oid, tb.oid, a.atttypid))::varchar[], ',')
        ELSE NULL
    END AS enum_values,
```

**To Reproduce**
1. Set up CockroachDB cluster
2. Set up basic Yii2 app
3. Make any model query (Yii2 will grab column info)

**Expected behavior**
No error.

**Environment:**
 - CockroachDB version: serverless
 - Server OS: Linux
 - Client app: fails in both PHP/Yii2 and in DBeaver

**Additional context**
Total failure of app.

Jira issue: CRDB-18329

**Comment by Sarke** [comment]
Possibly related? #67951

**Comment by Sarke** [comment]
I found that both of these work:

```sql
    CASE WHEN null = 'e'::char
        THEN array_to_string((SELECT array_agg(enumlabel) FROM pg_enum WHERE enumtypid = COALESCE(td.oid, tb.oid, a.atttypid))::varchar[], ',')
        ELSE NULL
    END AS enum_values,
```

```sql
    CASE WHEN COALESCE(td.typtype, tb.typtype, t.typtype) = 'e'::char
        THEN array_to_string((SELECT array_agg(enumlabel) FROM pg_enum WHERE enumtypid = null)::varchar[], ',')
        ELSE NULL
    END AS enum_values,
```

So removing the `COALESCE` from either the condition or the subquery works.  We don't use enums so we can work around this, but it's interesting to note that the condition is always false so the error must happen before the subquery is executed?

**Comment by mgartner** [comment]
This was fixed by #95234, which should be included in v23.1. Closing.
